src/lightning_utilities/core/enums.py

```diff
@@ -5,15 +5,27 @@
 from enum import Enum
 from typing import Optional
 
+from typing_extensions import Literal
+
 
 class StrEnum(str, Enum):
-    """Type of any enumerator with allowed comparison to string invariant to cases."""
+    """Type of any enumerator with allowed comparison to string invariant to cases.
+
+    >>> class MySE(StrEnum):
+    ...     t1 = "T-1"
+    ...     t2 = "T-2"
+    >>> MySE("T-1") == MySE.t1
+    True
+    >>> MySE.from_str("t-2") == MySE.t2
+    True
```

Comment:
Can you write regular tests instead of doctests?

Reply:
added